### PR TITLE
[Merged by Bors] - feat: multiplication by `closure {1}` in topological groups does not change closed and open subsets

### PR DESCRIPTION
**Pull request by sgouezel**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Algebra/Group/Basic.lean [hunk 1]
```diff
@@ -1726,6 +1782,57 @@ theorem exists_isCompact_isClosed_subset_isCompact_nhds_one
...
+  have A : (y * x⁻¹) • K ∈ 𝓝 y := by
+    rw [← preimage_smul_inv]
+    exact (continuous_const_smul _).continuousAt.preimage_mem_nhds (by simpa using h)
+  rcases exists_mem_nhds_isClosed_subset (inter_mem A hn) with ⟨L, hL, L_closed, LK⟩
+  refine ⟨L, hL, LK.trans (inter_subset_right _ _), ?_⟩
+  exact (hK.smul (y * x⁻¹)).of_isClosed_subset L_closed (LK.trans (inter_subset_left _ _))
+
+/-- A topological group which is weakly locally compact is automatically locally compact. -/
+@[to_additive]
+lemma WeaklyLocallyCompactSpace.locallyCompactSpace_of_group [WeaklyLocallyCompactSpace G] :
```
Comment (ocfnash) [inline Mathlib/Topology/Algebra/Group/Basic.lean:1800]: Can this be an instance?

**Review by ocfnash — APPROVED** [review]
Thanks!

I note in passing that `Mathlib.Topology.Algebra.Group.Basic` was already very long. If you have the appetite to follow up and split it, that would be great!

**Comment by ocfnash** [comment]
bors d+

File: Mathlib/Topology/Algebra/Group/Basic.lean (hunk 1 above)
Comment (sgouezel) [reply Mathlib/Topology/Algebra/Group/Basic.lean]: It creates a loop, but now loops are not supposed to be a problem. Let me try it and see if it breaks.

**Pull request closed without merging**